src/torrent.cpp

```diff
@@ -1384,16 +1382,89 @@ bool is_downloading_state(int const st)
 #endif
 
 #ifdef TORRENT_SSL_PEERS
+	void torrent::init_ssl(string_view cert)
+	{
+		// create the SSL context for this torrent. We need to
+		// inject the root certificate, and no other, to
+		// verify other peers against
+#if BOOST_VERSION >= 106400
+		std::unique_ptr<ssl::context> ctx(new ssl::context(ssl::context::tls));
+#else
+		std::unique_ptr<ssl::context> ctx(new ssl::context(ssl::context::tlsv12));
+#endif
+
```

Comment:
In the existing `torrent::init_ssl()` there was some error checking here if `!ctx`, which you removed. It doesn't seem like it would make sense to check that here though, so it should be fine to remove it. If the `context` constructor fails, it will throw, to make `operator new` return `nullptr`.